yt_dlp/extractor/dlf.py

```diff
@@ -0,0 +1,209 @@
+import re
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    extract_attributes,
+    int_or_none,
+    url_or_none,
+)
+
+
+class DLFBaseIE(InfoExtractor):
+    _VALID_URL_BASE = r'https?://(?:www\.)?deutschlandfunk\.de/'
+    _BUTTON_REGEX = r'(<button[^>]+alt="Anhören"[^>]+data-audio-diraid[^>]*>)'
+
+    def _parse_button_attrs(self, button, id=None):
+        attrs = extract_attributes(button)
+        id = id or attrs['data-audio-diraid']
+
+        # Note: DLF's mp3 recordings are in two distinct locations on the web:
+        # A direct download location, https://download.deutschlandfunk.de/ ... .mp3
+        # A stream with playback controls, https://ondemand-mp3.dradio.de/ ... .mp3
+        url = url_or_none(
+            attrs.get('data-audio-download-src')
+            or attrs.get('data-audio')
+            or attrs.get('data-audioreference')
+            or attrs['data-audio-src'])
+
```

Comment:
imo we can use a var to make things easier below
```suggestion
        ext = determine_ext(url)

```